### PR TITLE
Simplify and improve the build instructions for Menhir parsers.

### PR DESCRIPTION
**Pull request by fpottier**
These changes have the effect of activating `--infer`.
This is strongly recommended (in general).
Furthermore, in the specific case of `annot_parser`, activating `--infer` works around a bug in OCaml 4.07-4.10, which causes a failure during the compilation of `js_of_caml-compiler`.
Thus, I would recommend merging at least the first two commits in this PR.

Best regards,
François.

### CODE REVIEW

File: compiler/lib/dune [hunk 1]
```diff
@@ -19,43 +19,32 @@
...
-   TUnknown)))
+   TUnknown
+  )
+)
 
 (rule
  (alias runtest)
  (deps js_parser.conflicts.reference js_parser.conflicts)
  (action
   (diff js_parser.conflicts.reference js_parser.conflicts)))
```
Comment (smorimoto) [inline compiler/lib/dune:42]: Perhaps we can just remove this one as well?

**Review by smorimoto — APPROVED** [review]

**Comment by smorimoto** [comment]
Could you please confirm this PR?

**Comment by smorimoto** [comment]
@hhugo Could you please confirm this PR?

**Comment by hhugo** [comment]
Thank you for your contribution

**Pull request merged**